src/c_wrapping.hpp

```diff
@@ -500,6 +500,14 @@ void read_data(hmat_matrix_t * matrix, hmat_iostream readfunc, void * user_data)
     hmat::MatrixDataUnmarshaller<T>(readfunc, user_data).read(hmi->engine().hmat);
 }
 
+// Read a full hmatrix
+template <typename T, template <typename> class E>
+hmat_matrix_t *read( hmat_iostream readfunc, void *data ) {
+    hmat_matrix_t *hmat = read_struct<T,E>( readfunc, data );
+    read_data<T,E>( hmat, readfunc, data );
+    return hmat;
+}
+
```

Comment:
keep it within chameleon (idem for write)

Reply:
Done